packages/flutter_tools/lib/src/version.dart

```diff
@@ -236,20 +236,33 @@ class FlutterVersion {
     final Duration frameworkAge = _clock.now().difference(localFrameworkCommitDate);
     final bool installationSeemsOutdated = frameworkAge > kVersionAgeConsideredUpToDate;
 
+    /// Gets whether or not there is a new version of Flutter available on the remote.
+    ///
+    /// Returns null if the cached version is out-of-date or missing, and we are
+    /// unable to reach the server to get the latest version.
     Future<bool> newerFrameworkVersionAvailable() async {
       final DateTime latestFlutterCommitDate = await _getLatestAvailableFlutterVersion();
 
       if (latestFlutterCommitDate == null)
-        return false;
+        return null;
 
       return latestFlutterCommitDate.isAfter(localFrameworkCommitDate);
     }
 
+    // Get whether there's a newer version on the remote. This only goes
+    // to the server if we haven't checked recently so won't happen often.
+    // Note: This must come before the stamp load below because it also reads/writes
```

Comment:
While working on the tests, I noticed that newerFrameworkVersionAvailable used to be called after we loaded the stamp, and then later on the stamp is saved. This meant the changes made in `newerFrameworkVersionAvailable` would be overwritten by this stale version in memory.

It's possible that this didn't matter before, but it does not (one of the tests does two checks, and the last server ping timestamp didn't persist between checks) so I had to move it up here. Another option would be passing `stamp` into `newerFrameworkVersionAvailable`. This way seemed simplest, but not sure I like that re-ordering these two lines breaks a test.

Reply:
Yeah let's pass the stamp to the function.

Reply:
Actually, this turns out to not quite fix it. Inside `_getLatestAvailableFlutterVersion` we have:

```dart
await versionCheckStamp.store(
  newTimeVersionWasChecked: _clock.now(),
  newKnownRemoteVersion: remoteFrameworkCommitDate,
);
```

So the version that was loaded outside is always going to be stale if we load it before we call this method. Feels a little wonky, but not sure how to fix in a way that doesn't involve threading the same stamp instance all the way through (including in the test calls). WDYT?

Reply:
Hmm.  Yeah, sounds like threading it all the way through is the way to go.